### ISSUE TITLE
AMF、SMF and UPF failed to connect to internet 

### ISSUE THREAD

**Issue by linin00** [Orange-OpenSource/towards5gs-helm#74]
I deployed 5GC by referring to this [article](https://github.com/Orange-OpenSource/towards5gs-helm/blob/main/docs/demo/Setup-free5gc-and-test-with-UERANSIM.md), but all the network elements except amf and smf started successfully. Checking found that both amf and smf are stuck in the initialization process because the process of initializing the container wait-nrf keeps curl http://nrf-nnrf:8000, but the **status code returned is 000**. I tried to ping google.com in amf, smf and upf pod by exec command, but it will show ping: **bad address 'google.com'**. It looks like all pods with multiple NICs deployed, their network functionality is broken, how can I solve this problem? Thanks!

![image](https://user-images.githubusercontent.com/66274243/223025666-a4a05a91-c154-4526-931f-ca23c55abf90.png)

**Comment by pinoOgni** [comment]
Hi @linin00  it happened to me too and I think it depends on the MACVLAN cni-plugin you have installed. In my case a wrong version was downloaded or maybe corrupted I don't know. Indeed I tried copying the macvlan binary from another cluster where free5gc was running and after deleting and reinstalling free5gc again the pods started working.

**Comment by sergirem** [comment]
Hi @pinoOgni , I had the same issue but I do not understand how did you fix this error

Could you please explain me how to solve that problem? 

Kind regards

**Comment by pinoOgni** [comment]
Hi @sergirem, I hope I remember correctly.

Basically I already had a working cluster on GCP and after creating the second one, that error appeared. After investigating, I had found that all the versions of the various components between the two clusters were identical. 

I found that the error was about the macvlan, so the idea was to copy the cni plugin macvlan from the working cluster (I seem to remember that I only took that but perhaps I had also copied the multus one) and bring it to the cluster where I had the problem . Basically I copied the macvlan binary of the `/opt/cni/bin directory.

I don't think it's the most correct solution but at that moment it was the only idea in mind and it went well.
